File: rb/features.rb
# KorpoPro SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module KorpoProFeatures
  def self.make_feature(name)
    case name
    when "base"
      KorpoProBaseFeature.new
    when "ratelimit"
      KorpoProRatelimitFeature.new
    when "retry"
      KorpoProRetryFeature.new
    when "test"
      KorpoProTestFeature.new
    when "timeout"
      KorpoProTimeoutFeature.new
    else
      KorpoProBaseFeature.new
    end
  end
end

File: py/features.py
# KorpoPro SDK feature factory

from feature.base_feature import KorpoProBaseFeature
from feature.test_feature import KorpoProTestFeature


def _make_feature(name):
    features = {
        "base": lambda: KorpoProBaseFeature(),
        "test": lambda: KorpoProTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
